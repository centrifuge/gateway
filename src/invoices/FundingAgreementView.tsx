import React from 'react';
import { Link } from 'react-router-dom';
import { Box, Button, Heading } from 'grommet';
import { LabelValuePair } from '../common/interfaces';
import { connect } from 'react-redux';
import { RequestState } from '../store/reducers/http-request-reducer';
import { DocumentResponseHeader, FunFundingResponse, FunFundingResponseData } from '../../clients/centrifuge-node';
import { getInvoiceById, resetGetInvoiceById } from '../store/actions/invoices';
import { getContacts, resetGetContacts } from '../store/actions/contacts';
import { RouteComponentProps, withRouter } from 'react-router';
import { LinkPrevious } from 'grommet-icons';
import routes from '../routes';
import { signFunding } from '../store/actions/funding';
import { InvoiceDetails } from './InvoiceDetails';
import { Invoice } from '../common/models/invoice';
import { Preloader } from '../components/Preloader';

type ConnectedFundingAgreementViewProps = {
  getInvoiceById: typeof getInvoiceById;
  resetGetInvoiceById: typeof resetGetInvoiceById;
  getContacts: typeof getContacts;
  signFunding: typeof signFunding;
  resetGetContacts: typeof resetGetInvoiceById;
  invoice: Invoice | null;
  header: DocumentResponseHeader | null,
  fundingAgreement: FunFundingResponseData | null,
  id: string | null,
  contacts?: LabelValuePair[];
  signingFunding: RequestState<FunFundingResponse>;
} & RouteComponentProps<{ id?: string }>;

export class FundingAgreementView extends React.Component<ConnectedFundingAgreementViewProps> {

  displayName = 'FundingAgreementView';

  componentDidMount() {
    if (this.props.match.params.id) {
      this.props.getContacts();
      this.props.getInvoiceById(this.props.match.params.id);
    }
  }


  signFundingAgreement = () => {
    const { id, fundingAgreement, header, signFunding } = this.props;
    signFunding({
      document_id: header!.document_id,
      agreement_id: fundingAgreement!.funding!.agreement_id,
      invoice_id: id!,
    });
  };

  componentWillUnmount() {
    this.props.resetGetContacts();
    this.props.resetGetInvoiceById();
  }

  render() {
    const { invoice, contacts, fundingAgreement, signingFunding } = this.props;
    if (!invoice || !contacts) {
      return <Preloader message="Loading"/>;
    }

    if (signingFunding && signingFunding.loading) {
      return <Preloader message="Approving funding agreement" withSound={true}/>;
    }

    const canApproveFunding = !(fundingAgreement && fundingAgreement.signatures);

    return (
      <>

        <Box pad={{ bottom: 'large' }}>
          <Box justify="between" direction="row" align="center">
            <Box direction="row" gap="small" align="center">
              <Link to={routes.invoices.index} size="large">
                <LinkPrevious/>
              </Link>

              <Heading level="3">
                Funding Agreement for Invoice #{invoice!.number}
              </Heading>
            </Box>
            <Box direction="row" gap="medium">
              <Button
                disabled={!canApproveFunding}
                onClick={this.signFundingAgreement}
                primary
                label="Approve"
              />
            </Box>
          </Box>

          <InvoiceDetails invoice={invoice} fundingAgreement={fundingAgreement} contacts={contacts}/>
        </Box>
      </>
    );

  }
}


const mapStateToProps = (state) => {
  return {
    ...(!state.invoices.getById.data ? {
      invoice: null,
      header: null,
      fundingAgreement: null,
      id: null,
    } : {
      invoice: state.invoices.getById.data.data,
      id: state.invoices.getById.data._id,
      fundingAgreement: state.invoices.getById.data.fundingAgreement,
      header: state.invoices.getById.data.header,
      signingFunding: state.funding.sign,
    }),
    contacts: state.contacts.get.data
      ? (state.contacts.get.data.map(contact => ({
        label: contact.name,
        value: contact.address,
      })) as LabelValuePair[])
      : undefined,
  };
};

export const ConnectedFundingAgreementView = connect(
  mapStateToProps,
  {
    getContacts,
    resetGetContacts,
    getInvoiceById,
    resetGetInvoiceById,
    signFunding,
  },
)(withRouter(FundingAgreementView));


