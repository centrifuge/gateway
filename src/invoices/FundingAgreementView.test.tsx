import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { FundingAgreementView } from './FundingAgreementView';
import { createStore } from 'redux';
import getRootReducer from '../store/reducers';
import { AxisTheme } from '@centrifuge/axis-theme';


const DynamicParamFundingAgreementView = FundingAgreementView as any;
const store = createStore(getRootReducer({}), { router: { location: { pathname: '/' } } });
describe('FundingAgreementView', () => {

  const methods = {
    getInvoiceById: jest.fn(() => {
    }),
    resetGetInvoiceById: jest.fn(() => {
    }),
    getContacts: jest.fn(() => {
    }),
    resetGetContacts: jest.fn(() => {
    }),
  };

  const defaultProps = {
    header: {
      document_id: 'document_id',
    },
    match: {
      params: {
        id: '3333',
      },
    },
  };


  it('Should not render an empty form', () => {
    const invoice = {
      currency: "EUR"
    };
    const contacts = [
      {
        label: 'Alice',
        value: '0x1111',
      },
      {
        label: 'Bob',
        value: '0x1112',
      },
    ];
    const bodyShallow = renderer.create(
      // TODO Test fails when using Axis Theme. Investigate!!!
      <AxisTheme>
        <BrowserRouter>
          <DynamicParamFundingAgreementView
            invoice={invoice}
            contacts={contacts}
            {...methods}
            {...defaultProps}

          />
        </BrowserRouter>
      </AxisTheme>,
    ).toJSON();
    expect(bodyShallow).toMatchSnapshot();
  });


  it('Should render the form with proper values for contact fields', () => {
    const invoice = {
      sender: '0x1111',
      recipient: '0x1111',
      payee: '0x11112',
      currency:"USD",
    };
    const contacts = [
      {
        label: 'Alice',
        value: '0x1111',
      },
      {
        label: 'Bob',
        value: '0x11112',
      },
    ];
    const bodyShallow = renderer.create(
      <AxisTheme>
        <BrowserRouter>
          <DynamicParamFundingAgreementView
            invoice={invoice}
            contacts={contacts}
            {...methods}
            {...defaultProps}

          />
        </BrowserRouter>
      </AxisTheme>,
    ).toJSON();
    expect(bodyShallow).toMatchSnapshot();
  });

  it('Should render the loading state', () => {
    const invoice = null;
    const contacts = [
      {
        label: 'Alice',
        value: '0x1111',
      },
      {
        label: 'Bob',
        value: '0x11112',
      },
    ];
    const bodyShallow = renderer.create(
      <AxisTheme>
        <BrowserRouter>
          <DynamicParamFundingAgreementView
            invoice={invoice}
            contacts={contacts}
            {...methods}
            {...defaultProps}

          />
        </BrowserRouter>
      </AxisTheme>,
    ).toJSON();
    expect(bodyShallow).toMatchSnapshot();
  });


});
