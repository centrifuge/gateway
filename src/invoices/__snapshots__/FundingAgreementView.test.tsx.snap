// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundingAgreementView Should not render an empty form 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eWRwfm"
>
  <div
    className="StyledBox-sc-13pk1d4-0 kBwxZQ"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <svg
        id="L7"
        viewBox="0 0 100 100"
        width="56px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.6,3.5C5.9,13.6-6.6,42.7,3.5,68.4c10.1,25.7,39.2,38.3,64.9,28.1l-3.1-7.9c-21.3,8.4-45.4-2-53.8-23.3 c-8.4-21.3,2-45.4,23.3-53.8L31.6,3.5z"
        >
          <animateTransform
            attributeName="transform"
            attributeType="XML"
            dur="2s"
            from="0 50 50"
            repeatCount="indefinite"
            to="360 50 50"
            type="rotate"
          />
        </path>
        <path
          d="M42.3,39.6c5.7-4.3,13.9-3.1,18.1,2.7c4.3,5.7,3.1,13.9-2.7,18.1l4.1,5.5c8.8-6.5,10.6-19,4.1-27.7 c-6.5-8.8-19-10.6-27.7-4.1L42.3,39.6z"
        >
          <animateTransform
            attributeName="transform"
            attributeType="XML"
            dur="1s"
            from="0 50 50"
            repeatCount="indefinite"
            to="-360 50 50"
            type="rotate"
          />
        </path>
        <path
          d="M82,35.7C74.1,18,53.4,10.1,35.7,18S10.1,46.6,18,64.3l7.6-3.4c-6-13.5,0-29.3,13.5-35.3s29.3,0,35.3,13.5 L82,35.7z"
        >
          <animateTransform
            attributeName="transform"
            attributeType="XML"
            dur="2s"
            from="0 50 50"
            repeatCount="indefinite"
            to="360 50 50"
            type="rotate"
          />
        </path>
      </svg>
    </div>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
    />
    <span
      className="StyledText-sc-1sadyjn-0 cTRjlw"
    >
      Loading
    </span>
  </div>
</div>
`;

exports[`FundingAgreementView Should render the form with proper values for contact fields 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eWRwfm"
>
  <div
    className="StyledBox-sc-13pk1d4-0 hOJKqu"
  >
    <div
      className="sc-bdVaJa iBjyoJ StyledBox-sc-13pk1d4-0 dgXubw"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 gQLnBt"
      >
        <a
          href="/invoices"
          onClick={[Function]}
          size="large"
        >
          <svg
            aria-label="LinkPrevious"
            className="StyledIcon-ofa7kd-0 cgOtHq"
            viewBox="0 0 24 24"
          >
            <path
              d="M2,12 L22,12 M13,3 L22,12 L13,21"
              fill="none"
              stroke="#000"
              strokeWidth="2"
              transform="matrix(-1 0 0 1 24 0)"
            />
          </svg>
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
        />
        <h3
          className="StyledHeading-sc-1rdh4aw-0 YlnAZ"
        >
          Funding Agreement for Invoice #
        </h3>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 ffeAql"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 IRSNj"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 ffeAql"
          />
          <div
            className="StyledBox-sc-13pk1d4-0 IRSNj"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ffeAql"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 clLqce"
              >
                <div
                  className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                >
                  <div
                    className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                  >
                    <label
                      className="StyledText-sc-1sadyjn-0 cTRjlw"
                    >
                      Invoice number
                    </label>
                    <p
                      className="StyledParagraph-tbetod-0 giEJAX"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 jlbBeA"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 ffeAql"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 cyRkfg"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ffeAql"
            >
              <h5
                className="StyledHeading-sc-1rdh4aw-0 gVzBCh"
                style={
                  Object {
                    "minWidth": "100px",
                  }
                }
              >
                Sender
              </h5>
            </div>
            <div
              className="StyledBox-sc-13pk1d4-0 IRSNj"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 Tiesv"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 clLqce"
                >
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Name
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Centrifuge ID
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      >
                        0x1111
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                />
                <div
                  className="StyledBox-sc-13pk1d4-0 clLqce"
                >
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Street 1
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Street 2
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        City
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        State
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Country
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        ZIP code
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
          />
          <div
            className="StyledBox-sc-13pk1d4-0 bsrxMj"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ffeAql"
            >
              <h5
                className="StyledHeading-sc-1rdh4aw-0 gVzBCh"
                style={
                  Object {
                    "minWidth": "100px",
                  }
                }
              >
                Recipient
              </h5>
            </div>
            <div
              className="StyledBox-sc-13pk1d4-0 IRSNj"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 Tiesv"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 clLqce"
                >
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Name
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                />
                <div
                  className="StyledBox-sc-13pk1d4-0 clLqce"
                >
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Street 1
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Street 2
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        City
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        State
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        Country
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                  />
                  <div
                    className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                  >
                    <div
                      className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                    >
                      <label
                        className="StyledText-sc-1sadyjn-0 cTRjlw"
                      >
                        ZIP code
                      </label>
                      <p
                        className="StyledParagraph-tbetod-0 giEJAX"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 jlbBeA"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 IRSNj"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 hroMnY"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ffeAql"
            >
              <h5
                className="StyledHeading-sc-1rdh4aw-0 gVzBCh"
                style={
                  Object {
                    "minWidth": "100px",
                  }
                }
              >
                Invoice Details
              </h5>
            </div>
            <div
              className="StyledBox-sc-13pk1d4-0 IRSNj"
            >
              <div
                className="StyledBox-sc-13pk1d4-0 IRSNj"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 kmJoJj"
                >
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Status
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                  />
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Currency
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        >
                          USD
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                  />
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Invoice date
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        >
                          
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                  />
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Date due
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        >
                          
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
                />
                <div
                  className="StyledBox-sc-13pk1d4-0 ffeAql"
                >
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Net amount
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        >
                          $NaN
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                  />
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Tax rate
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        >
                          NaN%
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                  />
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Tax amount
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        >
                          $NaN
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
                  />
                  <div
                    className="StyledBox-sc-13pk1d4-0 dRJqWv"
                  >
                    <div
                      className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
                    >
                      <div
                        className="DisplayField__LabelBox-lfek92-1 jtIOTF StyledBox-sc-13pk1d4-0 Thbsl"
                      >
                        <label
                          className="StyledText-sc-1sadyjn-0 cTRjlw"
                        >
                          Gross amount
                        </label>
                        <p
                          className="StyledParagraph-tbetod-0 giEJAX"
                        >
                          $NaN
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 jlbBeA"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 ffeAql"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 jywknh"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 ffeAql"
            >
              <h5
                className="StyledHeading-sc-1rdh4aw-0 gVzBCh"
                style={
                  Object {
                    "minWidth": "100px",
                  }
                }
              >
                Comments
              </h5>
            </div>
            <div
              className="StyledBox-sc-13pk1d4-0 IRSNj"
            >
              <div
                className="DisplayField__DisplayFieldBox-lfek92-0 bzzYlY StyledBox-sc-13pk1d4-0 IRSNj"
              >
                <p
                  className="StyledParagraph-tbetod-0 giEJAX"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FundingAgreementView Should render the loading state 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eWRwfm"
>
  <div
    className="StyledBox-sc-13pk1d4-0 kBwxZQ"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <svg
        id="L7"
        viewBox="0 0 100 100"
        width="56px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.6,3.5C5.9,13.6-6.6,42.7,3.5,68.4c10.1,25.7,39.2,38.3,64.9,28.1l-3.1-7.9c-21.3,8.4-45.4-2-53.8-23.3 c-8.4-21.3,2-45.4,23.3-53.8L31.6,3.5z"
        >
          <animateTransform
            attributeName="transform"
            attributeType="XML"
            dur="2s"
            from="0 50 50"
            repeatCount="indefinite"
            to="360 50 50"
            type="rotate"
          />
        </path>
        <path
          d="M42.3,39.6c5.7-4.3,13.9-3.1,18.1,2.7c4.3,5.7,3.1,13.9-2.7,18.1l4.1,5.5c8.8-6.5,10.6-19,4.1-27.7 c-6.5-8.8-19-10.6-27.7-4.1L42.3,39.6z"
        >
          <animateTransform
            attributeName="transform"
            attributeType="XML"
            dur="1s"
            from="0 50 50"
            repeatCount="indefinite"
            to="-360 50 50"
            type="rotate"
          />
        </path>
        <path
          d="M82,35.7C74.1,18,53.4,10.1,35.7,18S10.1,46.6,18,64.3l7.6-3.4c-6-13.5,0-29.3,13.5-35.3s29.3,0,35.3,13.5 L82,35.7z"
        >
          <animateTransform
            attributeName="transform"
            attributeType="XML"
            dur="2s"
            from="0 50 50"
            repeatCount="indefinite"
            to="360 50 50"
            type="rotate"
          />
        </path>
      </svg>
    </div>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 irlfVP"
    />
    <span
      className="StyledText-sc-1sadyjn-0 cTRjlw"
    >
      Loading
    </span>
  </div>
</div>
`;
